src/mlsquare/architectures/sklearn.py

```diff
@@ -85,6 +87,83 @@ def update_params(self, params):
     def adapter(self):
         return self._adapter
 
+class DimensionalityReductionModel:
```

Comment:
You needed 
`SkLearnTfMatrixDecompositions` as an adapter. 
and 
Define a new DimensionalityReductionModel class under architectures, which inherits from BaseModel. This should have its custom set_param or get_param.
When defining SVD class, right now it is inheriting from GLM. It is totally unnecessary.